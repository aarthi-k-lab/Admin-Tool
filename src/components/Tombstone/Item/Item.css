.item {
  border-right: dashed 0.0625rem var(--grey-350);
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;
  padding: 0 1.25rem;
  padding-bottom: 0.8125rem;
  padding-top: 0.75rem;
  width: -webkit-fill-available;
  vertical-align: center;
}

.title {
  align-self: auto;
  color: var(--grey-590);
  font-size: 0.9em;
  height: 1em;
  display: contents;
}

.content {
  color: var(--grey-850);
  font-size: 0.9em;
  font-weight: bold;
  height: 1em;
  text-transform: uppercase;
  display: block;
}
