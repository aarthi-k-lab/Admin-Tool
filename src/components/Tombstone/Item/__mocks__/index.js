const Item = jest.fn(() => null);
export default Item;
