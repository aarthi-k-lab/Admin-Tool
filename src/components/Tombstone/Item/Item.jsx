import React from 'react';
import PropTypes from 'prop-types';

import './Item.css';

function Item({ content, title }) {
  let contentStyle = 'content';
  if (title === 'Previous Disposition' && content.trim().toLowerCase() === 'send for qc review') {
    contentStyle = 'qCReviewContent';
  }
  return (
    <div styleName="item">
      <span styleName="title">{title}</span>
      <span styleName={contentStyle}>{content}</span>
    </div>
  );
}

Item.defaultProps = {
  content: 'content',
  title: 'title',
};

Item.propTypes = {
  content: PropTypes.string,
  title: PropTypes.string,
};

export default Item;
