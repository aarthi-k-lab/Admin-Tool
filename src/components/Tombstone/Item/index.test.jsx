import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Item from '.';

describe('<Item />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<Item />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('contains default title and content when no props are passed', () => {
    const wrapper = shallow(<Item />);
    expect(wrapper.find('div').containsAllMatchingElements([
      <span>title</span>,
      <span>content</span>,
    ])).toBe(true);
  });

  test('contains title and content which are passes as props', () => {
    const wrapper = shallow(<Item content="Loan" title="9876543" />);
    expect(wrapper.find('div').containsAllMatchingElements([
      <span>Loan</span>,
      <span>9876543</span>,
    ])).toBe(true);
  });
});
