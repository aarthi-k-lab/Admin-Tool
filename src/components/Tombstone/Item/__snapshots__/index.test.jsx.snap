// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> snapshot test 1`] = `
<div
  className="components-Tombstone-Item-Item__item"
>
  <span
    className="components-Tombstone-Item-Item__title"
  >
    title
  </span>
  <span
    className="components-Tombstone-Item-Item__content"
  >
    content
  </span>
</div>
`;
